### DISCUSSION TITLE
Generate "searchable" .PDF from .XPS

### DISCUSSION THREAD

**Discussion by cconur** [pymupdf/PyMuPDF#2397] [Q&A] [answered]
Hi there!
First of all big congrats for this great library.
I'm not able to make "searchable" a pdf file converted from an XPS input file. The result file is like an image, it is not possible to find any word at all. Should I include any special parameter?

This os the code I use:
```
import sys, fitz


# convert an XPS file to PDF
xps = fitz.open("input.xps")
pdfbytes = xps.convert_to_pdf()

pdfout = open("output.pdf", "wb")
pdfout.tobytes(pdfbytes)
pdfout.close()
```
Many thanks

**Comment by JorjMcKie** [comment]
No all looks good.
BTW you shouldn't even need to convert to PDF: search and text extraction also work on other MuPDF supported documents.
However, if the document is scanned images only, extraction will not work in either case.

**Reply by cconur** [reply]
Hi Jorj! many thanks for your answer. The point is that the XPS file is very large (900MB) and contains about 3000 pages (full of text and graphics). I would like to have it stored in pdf instead of XPS, but with the possibility to search any word within. What do you recommend?

**Reply by JorjMcKie** [reply, marked as answer]
First I would check if the XPS itself is searchable, respectively text is extractable. If not, then the conversion to PDF won't be either.
If you confirmed that your XPS file's text cannot be extracted / searched, you can convert it to PDF and then OCR that file.
BTW, an image-only document would be a good explanation for its huge size: my science magazines also are image-heavy but the average page size is about 150 KB and not over 300 KB like in your case.
